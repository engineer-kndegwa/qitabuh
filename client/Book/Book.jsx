import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

const Book = ({id, title, selectBook, isSelected, selectDuration }) => {
    let durationSection = null;
    if(isSelected){
        durationSection = <input 
        className="duration-selection"
        onChange={(e) => selectDuration(e, id)}
        type="number" 
        placeholder={`Enter duration in days to rent --> ${title}`}
        />
    };
    return <li className={classnames('book-ctn',
        {"selected": isSelected},
        )} onClick={(e) => selectBook(e, id, title)}>
        <label className='checkbox-ctn'>
         <input 
         type='checkbox'
         onChange={(e) => onChangeDuration(e)} checked={isSelected}/>
         <span className='checkbox-text'>{title}</span>
        </label>
        {durationSection}
    </li>  
}

Book.propTypes = {
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    selectBook: PropTypes.func.isRequired,
    isSelected: PropTypes.bool.isRequired,
    selectDuration: PropTypes.func.isRequired
};

export default Book;