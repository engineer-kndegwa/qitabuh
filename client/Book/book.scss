.book-ctn {
    padding: 1em;
    margin: 2em 0;
    cursor: pointer;
    box-shadow: 2px 2px 30px rgba($secondary-color, 0.2);
    transition: all .4s ease-in-out;
    display: flex;
    justify-content: space-between;
    align-items: center;

    span {
        width: 40%;
    }
    
    &.selected {
        background: $primary-color-2;
        color: $white;
    }

    .duration-selection{
        border: none;
        width: 50%;
        padding: 10px;
        font-size: 1em;

        &:focus {
            outline: 8px solid rgba($secondary-color-2, 0.3);
        }
    }
};