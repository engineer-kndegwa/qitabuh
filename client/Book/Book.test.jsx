import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

// Component
import Book from "./Book.jsx";

const dummyProps = {};

beforeAll(() => {
    const props = {
        id: 'abcd',
        title: "Title",
        selectBook: () => {},
        selectDuration: () => {},
        isSelected: true
    };
    
    const propsAlt = {
        id: 'efgh',
        title: "Title",
        selectBook: () => {},
        selectDuration: () => {},
        isSelected: false
    };

    dummyProps.props = props;
    dummyProps.propsAlt = propsAlt;
});

test('Should render correctly when there are no items', () => {
    const { props } = dummyProps;
    const tree = renderer.create(<Book {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
});

test('Should render duration selection input if isSelected is true', () => {
    const { props, propsAlt } = dummyProps;
    const wrapper = shallow(<Book {...props}/>);
    const wrapperAlt = shallow(<Book {...propsAlt}/>);
    expect(wrapper.containsMatchingElement(<input className="duration-selection"/>)).toBe(true);
    expect(wrapperAlt.containsMatchingElement(<input className="duration-selection"/>)).toBe(false);
});
