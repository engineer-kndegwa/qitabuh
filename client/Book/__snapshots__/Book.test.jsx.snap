// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly when there are no items 1`] = `
<li
  className="book-ctn selected"
  onClick={[Function]}
>
  <label
    className="checkbox-ctn"
  >
    <input
      checked={true}
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="checkbox-text"
    >
      Title
    </span>
  </label>
  <input
    className="duration-selection"
    onChange={[Function]}
    placeholder="Enter duration in days to rent --> Title"
    type="number"
  />
</li>
`;
